MyMovies/View Controllers/MyMoviesTableViewController.swift

```diff
@@ -7,40 +7,128 @@
 //
 
 import UIKit
+import CoreData
 
 class MyMoviesTableViewController: UITableViewController {
+    
+    //let movieController = MovieController()
+    
+    // MARK: - Properties
+      
+      lazy var fetchedResultsController: NSFetchedResultsController<Movie> = {
+          let fetchRequest: NSFetchRequest<Movie> = Movie.fetchRequest()
+          fetchRequest.sortDescriptors = [NSSortDescriptor(key: "title", ascending: true),
+                                         NSSortDescriptor(key: "hasWatched", ascending: true)]
+          
+          let context = CoreDataStack.shared.mainContext
+          let frc = NSFetchedResultsController(fetchRequest: fetchRequest, managedObjectContext: context, sectionNameKeyPath: "title", cacheName: nil)
+          
+          frc.delegate = self
+          
+          do {
+              try frc.performFetch()
+          } catch {
+              NSLog("Error performing initial fetch inside fetchedResultsController: \(error)")
+          }
+          return frc
+      }()
+    
+    
 
     override func viewDidLoad() {
         super.viewDidLoad()
+        tableView.delegate = self
+        tableView.reloadData()
     }
 
     // MARK: - Table view data source
 
     override func numberOfSections(in tableView: UITableView) -> Int {
         // #warning Incomplete implementation, return the number of sections
-        return 0
+        return fetchedResultsController.sections?.count ?? 1
+        
     }
 
     override func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
         // #warning Incomplete implementation, return the number of rows
-        return 0
+        return fetchedResultsController.sections?[section].numberOfObjects ?? 0
     }
     
-    /*
     override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
-        let cell = tableView.dequeueReusableCell(withIdentifier: "reuseIdentifier", for: indexPath)
-
-        // Configure the cell...
+        guard let cell = tableView.dequeueReusableCell(withIdentifier: MovieTableViewCell.reuseIdentifier, for: indexPath) as? MovieTableViewCell else {
+            fatalError("Can't dequeue cell of type \(MovieTableViewCell.reuseIdentifier)")
+        }
+        
 
+        cell.movie = fetchedResultsController.object(at: indexPath)
+        cell.movieController = MovieController()
```

Comment:
you should set this to the movieController you have commented out up top.
```suggestion
        cell.movieController = self.movieController
```